### PR TITLE
filtered curriculum-topics and curriculum-topics-folder from searchresults

### PR DESCRIPTION
**Pull request by Knorrke**

### CODE REVIEW

File: src/module/Taxonomy/src/Taxonomy/Filter/SearchableTaxonomyCollectionFilter.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        return $value->filter(
+            function (TaxonomyTermInterface $term) {
+                $type = $term->getType()->getName();
+                return  $type != 'curriculum-topic' && $type != 'curriculum-topic-folder';
+            }
+        );
+    }
+}
```
Comment (arekkas) [inline src/module/Taxonomy/src/Taxonomy/Filter/SearchableTaxonomyCollectionFilter.php]: Add linebreak :)

**Comment by Knorrke** [comment]
I don't really know if this works, couldn't test it (search doesn't work in my local vagrant). Is there a way to test it anyhow?

File: src/module/Taxonomy/src/Taxonomy/Filter/SearchableTaxonomyCollectionFilter.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            throw new Exception\RuntimeException(sprintf(
+                'Expected Collection but got %s',
+                is_object($value) ? get_class($value) : gettype($value)
+            ));
+        }
+
+        return $value->filter(
+            function (TaxonomyTermInterface $term) {
+                $type = $term->getType()->getName();
+                return  $type != 'curriculum-topic' && $type != 'curriculum-topic-folder';
```
Comment (arekkas) [inline src/module/Taxonomy/src/Taxonomy/Filter/SearchableTaxonomyCollectionFilter.php]: Could you make this configurable, e.g. by passing these values in the constructor and renaming it to "TaxonomyTypeCollectionFilter"

Comment (arekkas) [inline src/module/Taxonomy/src/Taxonomy/Filter/SearchableTaxonomyCollectionFilter.php]: Also please use `!==` for strict type checking

Comment (arekkas) [inline src/module/Taxonomy/src/Taxonomy/Filter/SearchableTaxonomyCollectionFilter.php]: An array would also simplify this logic: `in_array($type, $forbiddenTypes)`

**Comment by arekkas** [comment]
> I don't really know if this works, couldn't test it (search doesn't work in my local vagrant). Is there a way to test it anyhow?

Sure:
- Write unit tests
- Lucene *should* be installed on the local environment, but it is not fed with information in order to reduce VM load. Try running (in your vm): `php ./src/public/index.php search rebuild` (see [here](https://github.com/serlo-org/athene2/blob/master/src/module/Search/config/module.config.php#L68) ). It has been a long time since I tested solr on the local env, so it *might* not work, but it *could*. ;) Please be aware that rebuilding the search index might take some time

File: src/module/Taxonomy/src/Taxonomy/Filter/TaxonomyTypeCollectionFilter.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+class TaxonomyTypeCollectionFilter implements FilterInterface
+{
+
+    private $forbiddenTypes = [];
+
+    /**
+     * @param $forbiddenTypes
+     */
+    function __construct($forbiddenTypes) {
+        $this->$forbiddenTypes = $forbiddenTypes;
```
Comment (arekkas) [inline src/module/Taxonomy/src/Taxonomy/Filter/TaxonomyTypeCollectionFilter.php]: I don't think the `$` belongs to $this->**$**forbiddenTypes

File: src/module/Taxonomy/src/Taxonomy/Filter/TaxonomyTypeCollectionFilter.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            throw new Exception\RuntimeException(sprintf(
+                'Expected Collection but got %s',
+                is_object($value) ? get_class($value) : gettype($value)
+            ));
+        }
+
+        return $value->filter(
+            function (TaxonomyTermInterface $term) {
+                $type = $term->getType()->getName();
+                return !in_array($type, $this->$forbiddenTypes);
```
Comment (arekkas) [inline src/module/Taxonomy/src/Taxonomy/Filter/TaxonomyTypeCollectionFilter.php]: See above

**Comment by arekkas** [comment]
Please give feedback if you were able to positively test this, messing with search indizes can be really bad in production.

File: src/module/Search/src/Search/Provider/TaxonomyProvider.php [hunk 5]
```diff
@@ -93,8 +95,13 @@ public function provide()
     {
         $container = [];
         $terms     = $this->taxonomyManager->findAllTerms(true);
-        $filter    = new NotTrashedCollectionFilter();
-        $terms     = $filter->filter($terms);
+        $notTrashed = new NotTrashedCollectionFilter();
+        $typeFilter = new TaxonomyTypeCollectionFilter('curriculum-topic', 'curriculum-topic-folder');
```
Comment (arekkas) [inline src/module/Search/src/Search/Provider/TaxonomyProvider.php]: This does not look correct, you're not using a variadic function.

**Comment by Knorrke** [comment]
> Lucene should be installed on the local environment, but it is not fed with information in order to reduce VM load. Try running (in your vm): php ./src/public/index.php search rebuild (see here ). It has been a long time since I tested solr on the local env, so it might not work, but it could. ;) Please be aware that rebuilding the search index might take some time

thanks, this worked! I've now tested it and with the latest changes it does work.

**Pull request merged**